neo/Network/LocalNode.cs

```diff
@@ -284,25 +284,28 @@ private static void CheckMemPool()
             }
         }
 
-        private IPEndPoint[] GetIPEndPointsFromSeedList(int seedsToTake)
+        private IEnumerable<IPEndPoint> GetIPEndPointsFromSeedList(int seedsToTake)
         {
-            Random rand = new Random();
-            IPEndPoint[] endpoints = Settings.Default.SeedList
-                .OrderBy(p => rand.Next()).Take(seedsToTake)
-                .OfType<string>().Select(p => p.Split(':'))
-                .Select(p =>
+            if (seedsToTake > 0)
+            {
+                Random rand = new Random();
+                foreach (string hostAndPort in Settings.Default.SeedList.OrderBy(p => rand.Next()))
                 {
+                    if (seedsToTake == 0) break;
+                    string[] p = hostAndPort.Split(':');
+                    IPEndPoint seed;
                     try
                     {
-                        return GetIPEndpointFromHostPortAsync(p[0], int.Parse(p[1])).Result;
+                        seed = GetIPEndpointFromHostPortAsync(p[0], int.Parse(p[1])).Result;
```

Comment:
I didn't notice this in my first review of the commit, but this can actually return null, which will end up causing a NullReferenceException in the `connectToPeers(ipEndpoints)` call. Need to check for null and continue in that case also.